### PR TITLE
simplify fonts

### LINKED ISSUE #874: simplifying font by hand?
The fontforge script doesn't simplify glyphs even when `.simplify()` is called on each glyph or on a selection including all glyphs. Simplifying by hand gives these results:

- no simplifcation:
  - `greciliae.ttf`: 1047712 bytes
  - a small pdf including two scores : 21848 bytes
- simplified:
  - `greciliae.ttf`: 920228 bytes
  - a small pdf including two scores : 21512 bytes

which is 12% for the font and 1.5% for the pdf. Although the economy is quite small for the pdf, maybe it's still worth doing a simplification by hand for the different fonts we generate? It's fairly simple:

- open into fontforge
- select all glyphs (ctrl+a)
- Element->Simplify->Simplify
- File->Generate Font

What do you think about doing this before releasing the tarballs? Maybe not for betas and rcs (though it would be good to do it once before the final release to see if there is any problem)...

### PR DESCRIPTION
**Pull request by eroux**
fixes #874

### CODE REVIEW

**Comment by eroux** [comment]
With these simplifications, the output almost doesn't change: I think the changes are mostly due to anti-aliasing, but please double-check, but it doesn't reduce size as much as I hoped (`104k` -> `97k` instead of `92k`). I tried more aggressive simplification but the output was changed, sometime visibly (for pes deminutus mainly). Tell me what you think about the right balance, maybe more aggressive simplification is ok.

**Comment by henryso** [comment]
Are you sure about the numbers?  Greciliae is currently 1.1M.  Assuming you're off by an order of magnitude, a 40k reduction might be worth it.  Can you post some side-by-sides of the figures that change significantly with more aggressive simplification?

**Comment by eroux** [comment]
Sorry, I got it wrong by a factor 10: `1048040` -> `976544`. The most changing figure is pes deminutus:

diff-simpl

**Comment by henryso** [comment]
Can you post a side-by-side of the glyph?  I can see the difference in the contour you posted, but at a "normal" resolution, it might not matter all that much.

**Comment by eroux** [comment]
not simplified:

old

simplified:

new

**Comment by eroux** [comment]
Well... taking another look at it, I think both are fine (well... to say it another way: I cannot tell the difference) and it's probably a good idea to completely simplify the font?

**Comment by henryso** [comment]
It is very subtle, but I honestly like the non-simplified slope of the line better.  However, if I didn't have a side-by-side, I probably wouldn't notice.

File: fonts/simplify.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            sys.exit()
+    if len(args) == 0:
+        usage()
+        sys.exit(2)
+    fontfile = args[0]
+    font = fontforge.open(fontfile)
+    font.selection.all()
+    font.simplify(0.1, ('mergelines','ignoreslopes','setstarttoextremum'))
+    font.generate(fontfile)
+    font.close()
```
Comment (henryso) [inline fonts/simplify.py]: This technique doesn't work in `squarize.py`?

Comment (eroux) [inline fonts/simplify.py]: No, it produces a `1001528 B` font (I thought it did nothing but it seems I was wrong... it does less than this script though)

**Comment by eroux** [comment]
Well, if you see a difference side by side, let's not simplify further then... Gregorio is more about quality output than small pdf or small download size...

**Comment by eroux** [comment]
(let's wait for other comments)

**Comment by rpspringuel** [comment]
I have the same reaction as @henryso when seeing them side-by-side.

**Comment by eroux** [comment]
ok, I think this is ready to be merged then

**Comment by henryso** [comment]
No objections.  Merging.

**Pull request merged**